source/common/upstream/upstream_impl.cc

```diff
@@ -987,7 +988,18 @@ void MainPrioritySetImpl::updateCrossPriorityHostMap(const HostVector& hosts_add
   }
 
   for (const auto& host : hosts_removed) {
-    mutable_cross_priority_host_map_->erase(addressToString(host->address()));
+    auto existing_host = mutable_cross_priority_host_map_->find(addressToString(host->address()));
```

Comment:
nit: create a (const) temp var holding the `addressToString(host->address())` result, and pass it here and below.